subsys/net/ip/icmpv4.c

```diff
@@ -555,7 +555,24 @@ int net_icmpv4_send_echo_request(struct net_if *iface,
 	echo_req->sequence   = htons(sequence);
 
 	net_pkt_set_data(pkt, &icmpv4_access);
-	net_pkt_write(pkt, data, data_size);
+
+	if (data != NULL && data_size > 0) {
+		net_pkt_write(pkt, data, data_size);
+	} else if (data == NULL && data_size > 0) {
```

Comment:
Should we describe this special combo in the function documentation (same for the ipv6 version)

Reply:
I've extended the API docs to mention the new functionality.